org.adempiere.base.process/src/org/compiere/process/M_PriceList_Create.java

```diff
@@ -218,11 +218,13 @@ private String create() throws Exception {
 						+ " INNER JOIN M_Product p ON (p.M_Product_ID=po.M_Product_ID)"
 						+ " INNER JOIN M_DiscountSchemaLine dl ON (dl.M_DiscountSchemaLine_ID=?) "	//	#2
 						+ "WHERE p.AD_Client_ID IN (?, 0)"		//	#3
-						+ " AND p.IsActive='Y' AND po.IsActive='Y' AND po.IsCurrentVendor='Y'"
+						+ " AND p.IsActive='Y' AND po.IsActive='Y'"
 						//	Optional Restrictions
 						+ " AND (dl.Group1 IS NULL OR p.Group1=dl.Group1)"
 						+ " AND (dl.Group2 IS NULL OR p.Group2=dl.Group2)"
-						+ " AND (dl.C_BPartner_ID IS NULL OR po.C_BPartner_ID=dl.C_BPartner_ID)"
+						+ " AND (dl.VendorCategory IS NULL OR po.VendorCategory=dl.VendorCategory)"

```

Comment:
dl.VendorCategory should have no effect if dl.C_BPartner_ID > 0

Reply:
Of course there may be situations where all bpartner use the same vendorcategories. But normally you can only trust that a given vendorcategory hits the required products for a given bpartner. Therefore the normal usage would be to select a bpartner together with a vendorcategory.